nilearn/glm/tests/test_utils.py

```diff
@@ -31,12 +37,103 @@ def test_full_rank():
 
 
 def test_z_score():
-    p = np.random.RandomState(42).uniform(size=10)
-    assert_array_almost_equal(norm.sf(z_score(p)), p)
-    # check the numerical precision
-    for p in [1.e-250, 1 - 1.e-16]:
-        assert_array_almost_equal(z_score(p), norm.isf(p))
-    assert_array_almost_equal(z_score(np.float32(1.e-100)), norm.isf(1.e-300))
+    # ################# Check z-scores computed from t-values #################
+    # Randomly draw samples from the standard Student’s t distribution
+    tval = np.random.standard_t(10, size=10)
```

Comment:
please use a random number generator with a fixed seed

Reply:
also if 10 samples are drawn from the null hypothesis most likely there are no extreme values, so the precision issue doesn't appear? maybe add a few very large values to tval?

Reply:
Note that in this function we are not testing the problem of numerical accuracy addressed in this PR. This is a function to check whether the `z_score` function from `nilearn/_utils/glm.py` is working, regardless whether the corresponding p-values are extreme or not. I used size=10 because this was the length of the list of p-values used in the original test. Besides, note that this function had to be changed only because the `z_score` function was changed but its scope is still the same.